compiler/rustc_middle/src/query/mod.rs

```diff
@@ -118,6 +118,16 @@ rustc_queries! {
         desc { "perform lints prior to macro expansion" }
     }
 
+    /// Useful for the implementation of `std::env!`, `proc-macro`s change
```

Comment:
It should mention "tracked access to environment variables" as the first thing in the comment.